### ISSUE TITLE
Minor speech-related code errors

### ISSUE THREAD

**Issue by tomcw** [jeremysrand/a2bejwld#6]
The other night when I was checking your speech/SSI263 code, I spotted the following minor errors. Feel free to close this issue as "won't fix", as fixing them won't affect any current behaviour.

1) speech chip detection & start of a phrase: no need to write to DDRB,DDRA

From this line https://github.com/jeremysrand/a2bejwld/blob/bc1cbc975cfe7741e755b94b81cb2f14a8bf444e/a2bejwld/mockingboard_speech.s#L154
... there's no need to set $c403 & $c402 to zero.

and the same after this line: https://github.com/jeremysrand/a2bejwld/blob/bc1cbc975cfe7741e755b94b81cb2f14a8bf444e/a2bejwld/mockingboard_speech.s#L290

From this line https://github.com/jeremysrand/a2bejwld/blob/bc1cbc975cfe7741e755b94b81cb2f14a8bf444e/a2bejwld/mockingboard_speech.s#L238
... there's no need to restore $c403 & $c402 to $ff & $7 respectively (unless you are using the AY8910 sound chip).

and the same after this line https://github.com/jeremysrand/a2bejwld/blob/bc1cbc975cfe7741e755b94b81cb2f14a8bf444e/a2bejwld/mockingboard_speech.s#L397

All harmless though!

(In fact the writes to $c44n also get written to $c40n, so these 6522 DDRB,DDRA registers will get overwritten as soon as your speech interrupt starts writing to the SSI263 registers.)

2) speech chip detection: the 6522 interrupt is still latched

After detecting the speech chip, in the interrupt handler, you are missing a write to $c443 of $80 before this line:

https://github.com/jeremysrand/a2bejwld/blob/bc1cbc975cfe7741e755b94b81cb2f14a8bf444e/a2bejwld/mockingboard_speech.s#L228

ie. the code should be:
```
             lda   #$80  ; missing (set CTL=1)
             sta   $c443 ; missing
             lda   #$00
             sta   $c440
             lda   #$70  ; set CTL=0
             sta   $c443
```

Same here: https://github.com/jeremysrand/a2bejwld/blob/bc1cbc975cfe7741e755b94b81cb2f14a8bf444e/a2bejwld/mockingboard_speech.s#L378

Basically you are not setting the mode to DISABLE_IRQ (done by setting CTL=1, then CTL=0). so the speech chip will play a "pause" phoneme, then set IFR=$02 (as the SSI263 is still in interrupt mode). Since you have already cleared b1 in IER, then this won't trigger an interrupt on the 6502 (even when flag I=0).

But when you come to playback a phrase (eg. "Go") then when you set IER=$82 and CLI, the 6502 will _immediately_ get an interrupt (instead of waiting for 'pause' to complete before generating an interrupt). Your interrupt handler then correctly writes the next set of 5 SSI263 registers... so harmless again.

**Comment by jeremysrand** [comment]
Of these issues, I think the most serious is the interrupt still being latched after chip detection.  I originally grabbed this code from Total Replay and modified their code slightly.  It is possible I introduced a problem or perhaps they have/had this problem.  I will check their source code and see where the bug comes from.  One way or another, I plan to fix this one.

**Comment by jeremysrand** [comment]
I have merged the fixes that you reported to the TotalReplay project which is where much of the code you comment on came from.  Also, I fixed the other issues you reported with the speech code.  Thanks for the bug report!
